Sources/ScipioKit/Producer/PIF/XCBuildExecutor.swift

```diff
@@ -0,0 +1,182 @@
+import Logging
+import TSCBasic
+import PackageGraph
+import XCBuildSupport
+
+struct XCBuildExecutor {
+
+    var xcbuildPath: AbsolutePath
+
+    func build(
+        pifPath: AbsolutePath,
+        configuration: BuildConfiguration,
+        derivedDataPath: AbsolutePath,
+        buildParametersPath: AbsolutePath,
+        target: ResolvedTarget
+    ) async throws {
+        let executor = _Executor(args: [
+            xcbuildPath.pathString,
+            "build",
+            pifPath.pathString,
+            "--configuration",
+            configuration.settingsValue,
+            "--derivedDataPath",
+            derivedDataPath.pathString,
+            "--buildParametersFile",
+            buildParametersPath.pathString,
+            "--target",
+            target.name,
+        ])
+        try await executor.run()
+    }
+}
+
+private final class _Executor {
+    init(args: [String]) {
+        self.args = args
+
+        self.executor = ProcessExecutor<StandardErrorOutputDecoder>()
+
+        self.parser = XCBuildOutputParser(delegate: self)
+
+        executor.streamOutput = { [weak self] (bytes) in
+            self?.parser.parse(bytes: bytes)
+        }
+        executor.collectsOutput = false
+    }
+
+    let args: [String]
+
+    lazy var parser: XCBuildOutputParser = { preconditionFailure("uninitialized") }()
+    var executor: ProcessExecutor<StandardErrorOutputDecoder>
+
+    // FIXME: store log on file
+    var allMessages: [String] = []
+
+    var parseError: Error?
+    var executeError: Error?
+
+    var targets: [Int: XCBuildMessage.TargetStartedInfo] = [:]
+    var tasks: [Int: XCBuildMessage.TaskStartedInfo] = [:]
```

Comment:
どっちみちクラス自体が `private` ですが、これらもprivateにしてほしいです。

今のままだと、ファイル内からは変更可能なので

```suggestion
    private var allMessages: [String] = []

    private var parseError: Error?
    private var executeError: Error?

    private var targets: [Int: XCBuildMessage.TargetStartedInfo] = [:]
    private var tasks: [Int: XCBuildMessage.TaskStartedInfo] = [:]
```